src/AstConstructor.py

```diff
@@ -0,0 +1,71 @@
+from arpeggio import ParserPython, PTNodeVisitor, visit_parse_tree
+from AstNodes import *
+
+#This is the parser file for Smurf
+class NodeVisitor(PTNodeVisitor):
```

Comment:
Thought you'd fool me by putting `NodeVisitor` in a file called `AstConstructor.py`? Think again :)